javers-core/src/main/java/org/javers/core/metamodel/type/TypeFactory.java

```diff
@@ -126,13 +132,42 @@ JaversType inferFromAnnotations(Type javaType) {
         return create(builder.build());
     }
 
+    private ClientsClassDefinitionBuilder createShallowReferenceEntity(Class javaClass) {
+        Property idProperty = findIdProperty(javaClass);
+        if (idProperty == null) {
+            throw new JaversException(JaversExceptionCode.SHALLOW_REFERENCE_WITHOUT_ID, javaClass.getName());
+        }
+        return entityDefinition(javaClass)
+                .withIdPropertyName(idProperty.getName())
+                .withIgnoredProperties(getPropertyNamesWithoutIdProperty(javaClass));
+    }
+
     private boolean hasIdProperty(Class<?> javaClass) {
-        for (Property property : propertyScanner.scan(javaClass))  {
+        for (Property property : propertyScanner.scan(javaClass)) {
             if (property.looksLikeId()) {
                 return true;
             }
         }
         return false;
     }
+
+    private Property findIdProperty(Class<?> javaClass) {
```

Comment:
to jest CodeRepetition do EntityType.findDefaultIdProperty()
Kod TypeFactory był jak widać zbyt skomplikowany, więc trudno Ci było dodać zmianę z ShallowReference.
Umówmy się może tak, że cofniesz zmiany formatowania (które utrudniają Code Review) a ja zrefaktoruję i uproszczę kod TypeFactory, ok?